dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/ValidationUtils.java

```diff
@@ -245,18 +250,82 @@ public static <T extends ValueTypedDimensionalItemObject> void validateOptionSet
     }
   }
 
-  /**
-   * Check if the given UID has a valid format.
-   *
-   * @param checkUid a UID to be checked
-   * @param reporter a {@see Reporter} to which the error is added
-   * @param dto the dto to which the report will be linked to
-   * @param args list of arguments for the Error report
-   */
-  public static void checkUidFormat(
-      String checkUid, Reporter reporter, TrackerDto dto, Object... args) {
-    if (!CodeGenerator.isValidUid(checkUid)) {
-      reporter.addError(dto, ValidationCode.E1048, checkUid, args[0], args[1]);
+  public static void validateValueType(
+      @Nonnull Reporter reporter,
+      @Nonnull TrackerBundle bundle,
+      @Nonnull TrackerDto dto,
+      @Nonnull String value,
+      @Nonnull ValueTypedDimensionalItemObject object) {
+    ValueType valueType = object.getValueType();
+    String error;
+    if (valueIsValid(value, valueType) != null) {
+      error = String.format("Value type is %s but the value `%s` is not.", valueType, value);
+    } else {
+      error =
+          switch (valueType) {
+            case FILE_RESOURCE ->
+                bundle.getPreheat().get(FileResource.class, value) == null
+                    ? "File resource `" + value + "` could not be found."
+                    : null;
+            case IMAGE ->
+                bundle.getPreheat().get(FileResource.class, value) == null
+                    ? "File resource `" + value + "` could not be found."
+                    : validateImage(bundle, value);
+            case ORGANISATION_UNIT ->
+                bundle.getPreheat().getOrganisationUnit(value) == null
+                    ? "Organisation unit `" + value + "` could not be found."
+                    : null;
+            case USERNAME ->
+                bundle.getPreheat().getUserByUsername(value).isEmpty()
+                    ? "Username `" + value + "` could not be found."
+                    : null;
+            default -> null;
+          };
+    }
+
+    if (error != null) {
+      if (object instanceof TrackedEntityAttribute) {
+        reporter.addError(dto, ValidationCode.E1007, valueType, error);
+      } else if (object instanceof DataElement) {
+        reporter.addError(dto, ValidationCode.E1302, valueType, error);
+      }
+    }
+
+    if (valueType.isFile()) {
+      validateFileNotAlreadyAssigned(reporter, bundle, dto, value);
+    }
+  }
+
+  private static String validateImage(TrackerBundle bundle, String value) {
+    FileResource fileResource = bundle.getPreheat().get(FileResource.class, value);
+
+    return Constant.VALID_IMAGE_FORMATS.contains(fileResource.getFormat())
+        ? "File resource with uid '" + value + "' is not a valid image"
```

Comment:
We could be more specific saying what format this file resource has and what the valid ones are. That helps the user fix their issue 😄